File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for n in nums:
            i = abs(n) - 1
            nums[i] = -1 * abs(nums[i])

        return [i + 1 for i, val in enumerate(nums) if val > 0]

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> res = new ArrayList<>();
        for (int n : nums){
            int index = Math.abs(n) - 1;
            if (nums[index] > 0) { nums[index] = -nums[index]; }
        }

        for (int i = 0; i < nums.length; i++){
            if (nums[i] > 0) { res.add(i+1); }
        }
        return res;
    }
}